### PR TITLE
refactor spotify connection and add recommendation per track feature

### PR DESCRIPTION
**Pull request by xavluiz**
spotify connection updates
- ignore setting the name and jwt on successful connection as this will be set by the anon creation use case or the code time existance user use case
- add signup and login button after successful connection if not registered
- fetch spotify integration for a legacy connected spotify user

New feature
- Show recommendations based on a specific track selection

Clean up to prevent circular dependencies

### CODE REVIEW

File: src/LiveshareManager.ts [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
+import { getItem } from "./managers/FileManager";
 
 export async function manageLiveshareSession(session) {
     softwarePost("/data/liveshare", session, getItem("jwt"))
         .then(async resp => {
-            if (isResponseOk(resp)) {
-                logIt("completed liveshare sync");
-            } else {
-                logIt(`unable to sync liveshare metrics: ${resp.message}`);
+            if (!isResponseOk(resp)) {
```
Comment (brettmstevens7) [inline src/LiveshareManager.ts:7]: this function needed anymore?

File: src/Util.ts [hunk 2]
```diff
@@ -681,9 +563,39 @@ export function launchWebUrl(url) {
...
+  if (!isRegistered) {
+    return;
+  }
   open(`${launch_url}/music`);
 }
 
+export function checkRegistration(showSignup = true) {
+  if (!getItem("name")) {
+    if (showSignup) {
+      showModalSignupPrompt("A registered account is required to continue. Sign up or log in to continue.");
```
Comment (brettmstevens7) [inline src/Util.ts:576]: Sign up or register for a web.com account at Software.com to view your most productive music.

File: src/extension.ts [hunk 3]
```diff
@@ -30,23 +28,23 @@ export async function activate(ctx: ExtensionContext) {
...
 
   //
   // add the player commands before we show the playlist
   //
   ctx.subscriptions.push(createCommands());
 
-  // init the music manager and cody config
-  const musicMgr: MusicManager = MusicManager.getInstance();
+  // migrate legacy spotify access token info to integration info
+  await migrateAccessInfo();
```
Comment (brettmstevens7) [inline src/extension.ts:39]: Any potential unintended effects of doing the migration here vs. on the server?

File: src/managers/IntegrationManager.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  return false;
+}
+
+async function updateIntegrations(user, name) {
+  let foundNewIntegration: boolean = false;
+  let currentIntegrations = getIntegrations();
+  for (const integration of user.integrations) {
+    // {access_token, name, plugin_uuid, scopes, pluginId, authId, refresh_token, scopes}
+    if (integration.name.toLowerCase() === name && integration.status.toLowerCase() === "active") {
```
Comment (brettmstevens7) [inline src/managers/IntegrationManager.ts:25]: could you simplify this nesting a little bit if you did something like:
```javascript
const isActive = integration.name.toLowerCase() === name && integration.status.toLowerCase() === "active";
const isFound = currentIntegrations.find((n) => n.authId === integration.authId;
if (isActive && !isFound) {
   // do all the things
}
```

File: src/LiveshareManager.ts (hunk 1 above)
Comment (xavluiz) [reply src/LiveshareManager.ts:7]: no, not really. i was thinking of removing it as well. i'll get rid of it :)

File: src/Util.ts (hunk 2 above)
Comment (xavluiz) [reply src/Util.ts:576]: ok, cool. will update it to that

File: src/extension.ts (hunk 3 above)
Comment (xavluiz) [reply src/extension.ts:39]: i don't think so. we have the integration like this in the backend already

File: src/managers/IntegrationManager.ts (hunk 4 above)
Comment (xavluiz) [reply src/managers/IntegrationManager.ts:25]: yeah, totally. easier to read. i'll update it to that

Comment (xavluiz) [reply src/managers/IntegrationManager.ts]: updated

File: src/Util.ts (hunk 2 above)
Comment (xavluiz) [reply src/Util.ts]: updated, thanks

File: src/LiveshareManager.ts (hunk 1 above)
Comment (xavluiz) [reply src/LiveshareManager.ts]: removed

**Pull request merged**